tools/lint.py

```
@@ -909,9 +943,98 @@ def _within(path: Path, root: Path) -> bool:
+    ]
+    if not commands:
+        findings.append(
+            "delivery: hooks/hooks.json declares no SessionStart command"
+        )
+    for command in commands:
+        for ref in PLUGIN_ROOT_REF.findall(command):
+            if not (root / ref).exists():
+                findings.append(
+                    f"delivery: hooks/hooks.json names '{ref}', which does "
+                    f"not exist in the shipped tree"
+                )
```

Ruff (0.16.1)

[warning] Use `list.extend` to create a transformed list

(PERF401)